src/microbe_stage/editor/PatchMapDrawer.cs

```diff
@@ -752,7 +752,6 @@ private void RebuildMapNodes()
         }
 
         UpdateNodeSelections();
-        NotifySelectionChanged();
```

Comment:
Shouldn't this try to detect if the rebuild caused the selection to change, and if it did then run this method?
I worry about potential hard to debug bugs in the future popping up because of this when we have non-static maps.

Reply:
But how to?

Reply:
I think doing something like looking at the existing data, storing the selected patch if there is one. Then here check the selected patch again, and if it doesn't match the earlier value, then call this method. If there isn't already one a method to find the currently selected patch is needed to reduce code duplication.

Reply:
There's already a variable tracking selected patch.

Reply:
Then it should be made sure that that variable works correctly when the drawer is cleared and a change notification is sent correctly here.

Reply:
I am sure that NotifySelectionChanged shouldn't be called here.

Reply:
Let's consider the following (future scenario):
- The player has selected a patch in a map
- The GUI shows them the patch details etc.
- Now an event happens which changes the map so rebuild map is called. This causes the selected patch to be removed from the map.
- If the selection change notify callback is not called the other GUI state is incorrect.
That's why I think it is necessary in some cases to trigger NotifySelectionChanged when rebuilding the map.

Reply:
If the selected patch is removed, `SelectedPatch` property must be reassigned, which triggers `NotifySelectionChanged`

Reply:
I could not find any code which sets SelectedPatch to null, leading me to believe that the map rebuild will not function correctly in the scenario I outlined above.

Reply:
We don't have this scenario currently either.

Reply:
I'm bringing this up because removing this code function call here leaves the potential for a future bug, which will be harder than necessary to figure out. That's why I think if this method call is removed now, the related potential bug should also be fixed now.

Reply:
What's more, the code `NotifySelectionChanged` leads to reads `SelectedPatch` property. So if you leave it unchanged, even if you call `NotifySelectionChanged` here GUI will still be in the incorrect state.

Reply:
Yes, but my point being that seeing the NotifySelectionChanged call here, makes the person in the future changing the map rebuild logic more likely to do it correctly.

Reply:
Let's comment this out (

Reply:
I went ahead and wrote the right approach myself.

Reply:
I still doubt if that line will ever be called

Reply:
It might not. But maybe once we have plate tectonics simulation and the patch map is shown in the auto-evo exploring tool, then I foresee this being able to happen.